x-pack/plugins/cases/server/client/cases/utils.ts

```diff
@@ -160,75 +145,80 @@ const addAlertMessage = (
 };
 
 export const createIncident = async ({
-  actionsClient,
   theCase,
   userActions,
   connector,
-  mappings,
   alerts,
   casesConnectors,
   userProfiles,
-}: CreateIncidentArgs): Promise<MapIncident> => {
-  const {
-    comments: caseComments,
-    title,
-    description,
-    created_at: createdAt,
-    created_by: createdBy,
-    updated_at: updatedAt,
-    updated_by: updatedBy,
-  } = theCase;
-
-  const params = { title, description, createdAt, createdBy, updatedAt, updatedBy };
+  publicBaseUrl,
+}: CreateIncidentArgs): Promise<ExternalServiceIncident> => {
   const latestPushInfo = getLatestPushInfo(connector.id, userActions);
   const externalId = latestPushInfo?.pushedInfo?.external_id ?? null;
-  const defaultPipes = externalId ? ['informationUpdated'] : ['informationCreated'];
-  let currentIncident: ExternalServiceParams | undefined;
 
   const externalServiceFields =
     casesConnectors.get(connector.actionTypeId)?.format(theCase, alerts) ?? {};
 
-  let incident: Partial<PushToServiceApiParams['incident']> = { ...externalServiceFields };
+  const connectorMappings = casesConnectors.get(connector.actionTypeId)?.getMapping() ?? [];
+  const descriptionWithKibanaInformation = addKibanaInformationToDescription(
+    theCase,
+    userProfiles,
+    publicBaseUrl
+  );
 
-  if (externalId) {
-    try {
-      currentIncident = (await actionsClient.execute({
-        actionId: connector.id,
-        params: {
-          subAction: 'getIncident',
-          subActionParams: { externalId },
-        },
-      })) as unknown as ExternalServiceParams | undefined;
-    } catch (ex) {
-      throw new Error(
-        `Retrieving Incident by id ${externalId} from ${connector.actionTypeId} failed with exception: ${ex}`
-      );
-    }
-  }
+  const comments = formatComments({ userActions, latestPushInfo, theCase, userProfiles });
 
-  const fields = prepareFieldsForTransformation({
-    defaultPipes,
-    mappings,
-    params,
-  });
+  const mappedIncident = mapCaseFieldsToExternalSystemFields(
+    { title: theCase.title, description: descriptionWithKibanaInformation },
+    connectorMappings
+  );
 
-  const transformedFields = transformFields<BasicParams, ExternalServiceParams, Incident>({
-    params,
-    fields,
-    currentIncident,
-    userProfiles,
-  });
```

Comment:
I removed the unnecessary transformation of fields. We always overwrite the content with our values. If we want to support different actions in the feature we can code it better (I coded this hahahaha. I don't like it 😄 ).